### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by goranalkovic-infinum**
This PR brings a bunch of UX and visual improvements to the editor and our components.

# Change summary

## Responsive
- redesigned to make multiple stacked `responsive` components and their children easier to tell apart
- expanded state now groups everything visually
- when expanded the header is now sticky
- restyled expand button with new icon
- works without any code changes

## Other improvements
- removed separator from CustomSelect
- OptionPicker has 2 new params: `isToggleButtonActive` and `showToggleButtonLabel`
- PanelBody can now show icon on the left of the panel title, just need to add `className='es-panel-title'` and the title needs to be wrapped in a `<span>` (important!)
- If a ToggleControl is wrapped with a div that has a `className='es-icon-toggle es-icon-toggle--reverse'` and an icon is inserted before the ToggleControl, the controls will now render in a nice format
- _How to use X_ button in the HelpModal has a new style with a new icon
- added 8 new icons and fixed issues with a couple of existing ones
- helper illustrations now use the selected editor accent color for highlights
- control labels with icons are centered and spaced properly now, also icons are a bit bigger and take the editor accent color (fallback is default blue)
- Placeholder has a cleaner style and some spacing issues are fixed.

### CODE REVIEW

File: scripts/components/custom-select/custom-select.js [hunk 1]
```diff
@@ -141,6 +141,7 @@ export const CustomSelect = (props) => {
 					MultiValueLabel: SortableMultiValueLabel,
 					Option: customOptionComponent ?? Option,
 					SingleValue: customSingleValueDisplayComponent ?? SingleValue,
+					IndicatorSeparator: () => null,
```
Comment (dingo-d) [inline scripts/components/custom-select/custom-select.js:144]: Does this have to be a callback that isn't doing anything?

Comment (goranalkovic-infinum) [reply scripts/components/custom-select/custom-select.js:144]: Nope, copied it from the documentation 😅 
I'll change it

File: styles/override-editor.scss [hunk 2]
```diff
@@ -123,21 +123,68 @@ body {
...
+				margin-bottom: 0 !important;
+
+				&:first-of-type {
+					transition: 0.3s padding-left ease-out, 0.3s margin-left ease-out, 0.3s border-left-width ease-out;
+				}
+
+				&:first-of-type:not(.components-base-control__label) {
+					margin-left: 0.75rem;
+					padding-left: 1.25rem;
+					border-left: 1px solid hsl(0, 0%, 94%);
```
Comment (dingo-d) [inline styles/override-editor.scss:162]: Why `hsl`?

File: styles/override-editor.scss [hunk 3]
```diff
@@ -123,21 +123,68 @@ body {
...
-				margin-bottom: 0;
+		&.is-open {
+			padding: 0 0.5rem;
+
+			> .components-base-control__field {
+				padding: 0 0.75rem 0.75rem;
+				border-color: var(--wp-admin-theme-color, #111111);
+
+				> .components-base-control__label {
+					background-color: hsl(0, 0%, 100%);
```
Comment (dingo-d) [inline styles/override-editor.scss:175]: Same question here as the above one.

File: scripts/components/responsive/responsive.js [hunk 4]
```diff
@@ -1,8 +1,10 @@
 import React from 'react';
 import { __ } from '@wordpress/i18n';
 import { Fragment, useState } from '@wordpress/element';
-import { BaseControl, Button, Tooltip } from '@wordpress/components';
-import { aspectRatio } from '@wordpress/icons';
+import {
+	BaseControl, Button
```
Comment (gabriel-glo) [inline scripts/components/responsive/responsive.js:5]: This doesn't need to be in a separate line.

File: styles/override-editor.scss (hunk 2 above)
Comment (goranalkovic-infinum) [reply styles/override-editor.scss:162]: Cause I don't have to worry about calculating (percentage I want * 255)

File: styles/override-editor.scss [hunk 5]
```diff
@@ -255,4 +306,258 @@ body {
...
+			}
+		}
+	}
+
+	// Input error text style.
+	.es-input-error {
+		color: $base-error-color;
+		font-size: 0.85rem;
+		font-weight: 500;
+	}
```
Comment (dingo-d) [inline styles/override-editor.scss:562]: This whole file should be renamed to `editor-shame.scss` 😂

**Review by dingo-d — CHANGES_REQUESTED** [review]
Left a few questions, all looks good to me 👍🏼

**Review by dingo-d — APPROVED** [review]

File: styles/override-editor.scss (hunk 3 above)
Comment (gabriel-glo) [inline styles/override-editor.scss:175]: This is `#FFFFFF` and you are using `rgba()` function on the next line. 
I don't think mixing different color functions is beneficial to the code consistency.

**Review by gabriel-glo — CHANGES_REQUESTED** [review]

**Review by dingo-d — APPROVED** [review]

**Review by iruzevic — APPROVED** [review]

**Review by IvanGrginovInf — APPROVED** [review]

**Pull request merged**